src/graph/dfa.py

```diff
@@ -55,10 +55,10 @@ def _construct_dfa(self, states, edges):
             self.add_state(_s)
 
             # add init and accepting node attribute
-            if _s == "T0_init":
+            if _s.endswith("init"):
                 self.add_initial_state(_s)
 
-            if _s == "accept_all":
+            if _s.startswith("accept"):
```

Comment:
For now, I think this is fine. In the future, we should keep the constant parameters INIT and ACCEPT in one place and use them to check these lines.